spec/controllers/api/documents_controller_spec.rb

```diff
@@ -7,37 +7,67 @@
   let(:documents) { response_body['documents'] }
   let(:meta_data) { response_body['meta'] }
   let(:url) { "/api/en/documents#{params}" }
+  let(:review_layout)  { create :layout, identifier: 'review' }
+  let(:insight_layout)  { create :layout, identifier: 'insight' }
+  let!(:insight_page) { create(:insight_page_about_financial_wellbeing, site: site, layout: insight_layout) }
+  let!(:insight_page_1) { create(:insight_page_about_debt, site: site, layout: insight_layout) }
+  let!(:insight_page_2) { create(:insight_page_about_pensions, site: site, layout: insight_layout) }
+  let!(:review_page_1) { create(:page, site: site, layout: review_layout) }
 
   before do
     allow_any_instance_of(PageSerializer)
       .to receive(:related_content).and_return({})
+
+    get url
```

Comment:
@margOnline I think we can do something like this:

```ruby
let(:url) { '/api/en/documents' }

before do
  ...
  get url, params: params
```

I will verify this locally. This is a much better way though. Wdyt?

Reply:
In order to take this approach, the params need to be defined as hashes in the individual context blocks which is fine for something like 
`let(:params) { {document_type: 'insight', keyword: 'pension' } }` 
but for a phrase like 'Financial well being: the employee view', it has to be passed like as

```
{
  keyword: 'Financial well being: the employee view'
}

 whereas in the url it would be passed as: `Financial+well+being%3A+the+employee+view` so the former nicely tests that the above param is correctly converted to 'Financial well being: the employee view'

Reply:
@margOnline you will eventually have to define _params_ in the _context_ for every example block as a hash rather than as a text.

Reply:
regards @Tr4pSt3R's point, wdyt? @willhall88 @tomas-stefano

Reply:
I am not sure. I need to checked the code and test @Tr4pSt3R option to see if is better.

Reply:
I tested and Jones point makes sense. 

The code becomes more cleaner and the app parses right:

```
1890:Started GET "/api/en/documents?document_type=insight&keyword=Financial+well+being%3A+the+employee+view" for 127.0.0.1 at 2018-03-02 12:21:59 +0000
1892-Processing by API::DocumentsController#index as HTML
1893-  Parameters: {"document_type"=>"insight", "keyword"=>"Financial well being: the employee view", "locale"=>"en"}
```

The branch that I did the testing https://github.com/moneyadviceservice/cms/compare/8997-insight-keyword-search...testing-jones-idea

Reply:
👍